File: src/main/scala/uk/co/halfninja/randomnames/Name.scala
package uk.co.halfninja.randomnames

case class Name(val givenName: String, val familyName: String, val middleNames: Seq[String], val quotedName: String) {

	override def toString =
		(Seq(givenName) ++ 
		(Option(quotedName).map { n => '"' + n + '"' }.toSeq) ++
		middleNames ++
		Seq(familyName)).mkString(" ")
		
}

object Name {
	def apply(givenName: String, familyName: String) = new Name(givenName, familyName, Seq.empty, null)
}

File: src/main/kotlin/uk/co/halfninja/randomnames/Name.kt
package uk.co.halfninja.randomnames

data class Name(
    val givenName: String,
    val familyName: String,
    val middleNames: List<String> = emptyList(),
    val quotedName: String? = null
) {
    override fun toString(): String =
        listOfNotNull(
            givenName,
            quotedName?.let { "\"$it\"" },
            *middleNames.toTypedArray(),
            familyName
        ).joinToString(" ")
}
